### ISSUE TITLE
lisav2 should fail if the VM cannot be created/started on hyperv

### ISSUE THREAD

**Issue by ader1990** [LIS/LISAv2#391]
Currently, if the VM cannot be created or started on hyperv, the test cases will still be run.
As a consequence, valuable time is lost as all the test cases need to timeout and go to aborted state.

---

Additional behavior, merging issue #622 into here:

On Hyper-V, if the VM cannot be started - for example due to out of memory (on host), lisav2 will continue waiting for VM IP.
Current workflow:

[between tests] restore VM to snapshot
start VM
prints message that VM failed to start
enter in the "Waiting for IP address..." loop.
not continue, it will loop in the above.
Correct behavior: Start-VM should be handled and verified if it was able to start the VM, or catch the error and continue test run.

Messages for reference:

```
[INFO ] VM:vm_name restored to checkpoint: ICAbase
[hyperv] 'vm_name' failed to start.
[hyperv]
[hyperv] Not enough memory in the system to start the virtual machine vm_name.
[hyperv]
[hyperv] Could not initialize memory: Ran out of memory (0x8007000E).
[hyperv]
[hyperv] 'vm_name' failed to start. (Virtual machine ID x)
[hyperv]
[hyperv] Not enough memory in the system to start the virtual machine vm_name with ram size 4096 megabytes. (Virtual machine ID x)
[hyperv]
[hyperv] 'vm_name' could not initialize memory: Ran out of memory (0x8007000E). (Virtual machine ID x)
[INFO ] vm_name : Waiting for IP address...
[hyperv] [INFO ] vm_name : Waiting for IP address...
```

Related code should be here: https://github.com/LIS/LISAv2/blob/master/Libraries/HyperV.psm1#L542

**Comment by chvalean** [comment]
Should fail if VM create fails.
For VM start failures, that depends on the test case. We can either fallback to VM snapshot revert, and if it fails to starts even then, then go to exit.

**Comment by ader1990** [comment]
The problem is that we cannot fallback to a snapshot, as this issue appears when creating the VM before snapshotting it.

**Comment by juhlee-microsoft** [comment]
Close non-P0 issues

**Comment by chvalean** [comment]
Reopening this, @asofron is already working on a fix and this will be handled shortly.

**Comment by iamshital** [comment]
@asofron , How are you handling this error? are you going to wait and retry with timeout?
I'm also seeing another issues (but not consistently)-
1. Unable to snapshot (error unknown).
2. Unable to restore VM to earlier snapshot (error unknown).

**Comment by asofron** [comment]
@iamshital don`t intend to wait for retry and timeout ( too much time to aborted ) . As for the other two issues that your talking about are they happening after setup script ?

**Comment by iamshital** [comment]
@asofron Yes. Sometimes, it happened at creating the ICABase snapshot. and sometimes, at restoring ICABase snapshot. But again, repro is not consistent. So, I would put this low priority.

The important one is - How are you going to handle "Failed to start VM" error.

My suggestion is -
1. Delete the VMs if they are failed to start.

Scenario 1 : Single VM test - 
VM 1 fails to start -> proceed  to delete. (There is no use of keeping not started VMs hanging)

Scenario 2: Multiple VM test -
VM 1 starts, but VM 2 fails. -> proceed to delete VM1 and VM2 both. As there is no use of started VM also, if other VM is not started.
